src/containers/BooksForm.js

```diff
@@ -1,20 +1,82 @@
-import React from 'react';
+import React, { Component } from 'react';
+import { connect } from 'react-redux';
+import PropTypes from 'prop-types';
+import { createBook } from '../actions/index';
 
-const BooksForm = () => {
-  const categories = ['Action', 'Biography', 'History', 'Horror', 'Kids', 'Learning', 'Sci-Fi'];
+class BooksForm extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      title: '',
+      category: '',
+    };
+    this.baseState = this.state;
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
 
-  return (
-    <form>
-      <label htmlFor="title">
-        Title
-        <input id="title" type="text" name="title" />
-      </label>
-      <select id="category">
-        {categories.map(e => <option key={e} value={e}>{e}</option>)}
-      </select>
-      <button type="submit">Submit</button>
-    </form>
-  );
-};
+  async handleChange(e, field) {
+    await this.setState({
+      [field]: e.target.value,
+    });
```

Comment:
**OPTIONAL**: The `field` is already in `event.target` as `name`, so it might not be necessary to pass an extra `field` parameter.